src/hotspot/share/c1/c1_Compilation.hpp

```diff
@@ -130,6 +131,7 @@ class Compilation: public StackObj {
 
 
   static Compilation* current() {
+    DEBUG_ONLY(ciEnv::current()->check_compiler_data_c1_or_null();)
```

Comment:
So, can it be just some sort of:

```
    assert(CompilerThread::current()->compiler()->is_c1(), "sanity");
```

...without any other changes in `ciEnv`?

Reply:
Context: I think `ciEnv` is pretty compiler-agnostic, and it would be better to avoid exposing the fact C1/C2 exist to that interface, even if only for asserts. Seems cleaner to check that we are calling `Compilation::current()` from C1 and `Compile::current()` from C2.

Reply:
I agree with Aleksey here.

Reply:
Yes, I like this too. Simpler. Okay, I try that.